### PR TITLE
Merge 3.5.x up into 4.0.x

### PR DESCRIPTION
**Pull request by greg0ire**
@morozov can you please help with the failures?

```
1) Doctrine\DBAL\Tests\Platforms\MariaDBPlatformTest::testAlterTableColumnComments
Failed asserting that two arrays are equal.
--- Expected
+++ Actual
@@ @@
 Array (
-    0 => 'ALTER TABLE mytable ADD quota INT NOT NULL COMMENT 'A comment', CHANGE foo foo VARCHAR(255) NOT NULL, CHANGE bar baz VARCHAR(255) NOT NULL COMMENT 'B comment''
+    0 => 'ALTER TABLE mytable ADD quota INT NOT NULL COMMENT 'A comment', CHANGE foo foo VARCHAR(255) NOT NULL, CHANGE baz baz VARCHAR(255) NOT NULL COMMENT 'B comment''
 )

/home/greg/dev/doctrine-dbal/4.0.x/tests/Platforms/AbstractPlatformTestCase.php:412

2) Doctrine\DBAL\Tests\Platforms\MySQLPlatformTest::testAlterTableColumnComments
Failed asserting that two arrays are equal.
--- Expected
+++ Actual
@@ @@
 Array (
-    0 => 'ALTER TABLE mytable ADD quota INT NOT NULL COMMENT 'A comment', CHANGE foo foo VARCHAR(255) NOT NULL, CHANGE bar baz VARCHAR(255) NOT NULL COMMENT 'B comment''
+    0 => 'ALTER TABLE mytable ADD quota INT NOT NULL COMMENT 'A comment', CHANGE foo foo VARCHAR(255) NOT NULL, CHANGE baz baz VARCHAR(255) NOT NULL COMMENT 'B comment''
 )

/home/greg/dev/doctrine-dbal/4.0.x/tests/Platforms/AbstractPlatformTestCase.php:412

3) Doctrine\DBAL\Tests\Platforms\PostgreSQLPlatformTest::testAlterDecimalPrecisionScale
Failed asserting that two arrays are equal.
--- Expected
+++ Actual
@@ @@
 Array (
-    0 => 'ALTER TABLE mytable ALTER dloo1 TYPE NUMERIC(16, 6)'
-    1 => 'ALTER TABLE mytable ALTER dloo2 TYPE NUMERIC(10, 4)'
-    2 => 'ALTER TABLE mytable ALTER dloo4 TYPE NUMERIC(16, 8)'
+    0 => 'ALTER TABLE mytable ALTER dfoo1 TYPE NUMERIC(16, 6)'
+    1 => 'ALTER TABLE mytable ALTER dfoo2 TYPE NUMERIC(10, 4)'
+    2 => 'ALTER TABLE mytable ALTER dfoo4 TYPE NUMERIC(16, 8)'
 )
```

### CODE REVIEW

**Comment by morozov** [comment]
Most likely, those tests are the same invalid as the one removed in https://github.com/doctrine/dbal/pull/5623. I'll take a closer look later. IMO, unit tests that craft diffs manually are pointless and should be eventually removed.

**Comment by morozov** [comment]
@greg0ire the first test is definitely wrong as the previously removed one. If it saves you time, you can remove the failing tests in the merge commit and create an issue for 4.0.0 to revisit those tests and assign it to me. I will either reimplement them as functional or find the existing ones.

File: src/Platforms/SQLServerPlatform.php [hunk 1]
```diff
@@ -427,9 +427,11 @@ public function getAlterTableSQL(TableDiff $diff): array
             $requireDropDefaultConstraint = $this->alterColumnRequiresDropDefaultConstraint($columnDiff);
 
             if ($requireDropDefaultConstraint) {
+                $fromColumnName = $columnDiff->fromColumn->getName();
```
Comment (morozov) [inline src/Platforms/SQLServerPlatform.php]: The intermediate variable looks unnecessary.

**Review by morozov — APPROVED** [review]

**Comment by morozov** [comment]
It looks like some changes from #5622 got lost during the conflict resolution. By now, the `SchemaDiff` constructor should be marked as internal and the `$oldColumnName` property should be deprecated in `4.0.x`.

The latter is fine since I can just remove it but the `@internal` attribute needs to be carried over into future releases.

**Comment by greg0ire** [comment]
:thinking: that attribute seems to be present:

https://github.com/doctrine/dbal/blob/f84b9d92efd40ddf9a5809555b35fcd69cd793ec/src/Schema/SchemaDiff.php#L53

**Comment by morozov** [comment]
Sorry, I was referring to `ColumnDiff`: https://github.com/doctrine/dbal/blob/f84b9d92efd40ddf9a5809555b35fcd69cd793ec/src/Schema/ColumnDiff.php#L16-L23

**Pull request merged**